### PR TITLE
XD-3603: Port XD Log Sink (replace existing)

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/XD-3602

Initial s-c-s log sink is not configurable.

Make it compatible with the XD sink.

### CODE REVIEW

File: log-sink/pom.xml [hunk 1]
```diff
@@ -17,4 +17,15 @@
 		<start-class>org.springframework.cloud.stream.module.log.LogSinkApplication</start-class>
 	</properties>
 
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework.integration</groupId>
+			<artifactId>spring-integration-test</artifactId>
```
Comment (markfisher) [inline log-sink/pom.xml:23]: this should be test scope?

File: log-sink/src/main/java/org/springframework/cloud/stream/module/log/LogSink.java [hunk 2]
```diff
@@ -16,25 +16,33 @@
...
  * @author Marius Bogoevici
+ * @author Gary Russell
  */
 @EnableBinding(Sink.class)
+@EnableConfigurationProperties(LogSinkProperties.class)
 public class LogSink {
 
-	private static Logger logger = LoggerFactory.getLogger(LogSink.class);
+	@Autowired
+	private LogSinkProperties config;
```
Comment (markfisher) [inline log-sink/src/main/java/org/springframework/cloud/stream/module/log/LogSink.java]: we're still not 100% consistent across modules, but in most cases I think we're referring to the properties instance as `properties`

File: log-sink/src/main/java/org/springframework/cloud/stream/module/log/LogSinkProperties.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Gary Russell
+ *
+ */
+@ConfigurationProperties
+public class LogSinkProperties {
+
+	private String name = "log.sink"; // TODO: set from environment 'group' if available
```
Comment (markfisher) [inline log-sink/src/main/java/org/springframework/cloud/stream/module/log/LogSinkProperties.java]: in other cases, we've used `${spring.application.name}` (with a default like this value) for now, e.g.:

https://github.com/spring-cloud/spring-cloud-stream-modules/blob/master/hdfs-sink/src/main/java/org/springframework/cloud/stream/module/hdfs/sink/HdfsSinkProperties.java#L47

File: log-sink/pom.xml (hunk 1 above)
Comment (garyrussell) [inline log-sink/pom.xml:23]: oops; yes - it was automatically added by eclipse.

File: log-sink/src/main/java/org/springframework/cloud/stream/module/log/LogSink.java (hunk 2 above)
Comment (garyrussell) [inline log-sink/src/main/java/org/springframework/cloud/stream/module/log/LogSink.java]: I guess I picked a bad template :smile:  (FTP)

**Comment by garyrussell** [comment]
Pushed

**Comment by markfisher** [comment]
LGTM; merging

**Comment by markfisher** [comment]
squashed and merged as https://github.com/spring-cloud/spring-cloud-stream-modules/commit/ea2c770e5f4580eeecf1648d898f722744d98a74

File: log-sink/src/test/java/org/springframework/cloud/stream/module/log/LogSinkApplicationTests.java [hunk 4]
```diff
@@ -39,11 +52,39 @@
...
 
+	@SuppressWarnings("unused")
 	@Autowired
 	private Sink same;
 
+	@Autowired
+	private LoggingHandler logSinkHandler;
+
+	@BeforeClass
+	public static void setUp() {
```
Comment (ericbottard) [inline log-sink/src/test/java/org/springframework/cloud/stream/module/log/LogSinkApplicationTests.java:63]: Sorry, just looking at this now.
This `PropertiesInitializer` stuff should be replaced by `@[Web]IntegrationText({"name=foo", etc...})` IMHO

Comment (garyrussell) [inline log-sink/src/test/java/org/springframework/cloud/stream/module/log/LogSinkApplicationTests.java:63]: I was following the existing model in the FTP source.

If that model is incorrect, it should be fixed before asking "newbies" to contribute :smile:

**Pull request closed without merging**